lib/modules/merge.js

```diff
@@ -2,13 +2,24 @@ import { isPlainObject } from 'lodash/lang';
 import { isEmpty } from 'lodash/lang';
 
 export default function merge(state, stateDiff, options={}) {
-  const newState = { ...state };
-
   const {
     emptyDict='strict',
     array='replace',
   } = options;
 
+  // Empty state diffs get special handling. They shouldn't cause a new state
+  // object to be created in strict mode, but in replace mode we should return
+  // the new empty object.
+  if (!Object.keys(stateDiff).length) {
+    if (emptyDict === 'replace') {
+      return stateDiff;
+    }
+
+    return state;
```

Comment:
Can you elaborate on the difference between `skip` and `strict`? It seems like `skip` should be returning the old state, which makes sense. But should't `strict` be doing so as well? If both and `strict` and `skip` return state what I have seems fine